src/cursor/session.rs

```diff
@@ -184,6 +184,24 @@ where
         self.stream(session).next().await
     }
 
+    /// Update the type streamed values will be parsed as.
+    pub fn with_type<D>(mut self) -> SessionCursor<D>
+    where
+        D: DeserializeOwned + Unpin + Send + Sync,
+    {
+        let out = SessionCursor {
+            client: self.client.clone(),
+            info: self.info.clone(),
+            buffer: self.buffer.drain(..).collect(),
```

Comment:
I think [`std::mem::replace`](https://doc.rust-lang.org/std/mem/fn.replace.html) might be more efficient here.

Reply:
Good call (and clippy prompted me to replace that with `std::mem::take`).